### DISCUSSION TITLE
Get field validation state on `watch`

### DISCUSSION THREAD

**Discussion by kadoshms** [react-hook-form/react-hook-form#6295] [Q&A]
Hey!

As for https://github.com/react-hook-form/react-hook-form/discussions/6278. I am trying to create an interactive view that is directly affected by my form.

Long story short, when the form is updated, and valid, I execute some requests to the server and update the view accordingly.
Since it's impossible to subscribe to the value of `watch`, as it is not memoized, I did something as the following:

```
    useEffect(() => {
        const subscription = watch((data, options) => {
            const { name } = options;
            // if some field is changed, do the server request and update the view
        });

        return () => {
            subscription.unsubscribe();
        };
    }, [watch]);
```

Which works almost as expected, except of the fact that at this point, I can't access the validation state.
From what I saw, and debugged, the validation state of the form at the point of the callback is not updated yet, and only updates on next render.

I believe this is something that can be super useful. In fact, relying on the values from the form and it's state for other purposes other than "simply a form" sounds like a common use case.

I wonderd, if there's anyway to run the validation function at the point of the watch callback, or even better - get the next validation state within the callback itself.
Supporting this will open this ability for great stuff.

I'd appreciate your feedback on this one.

Thanks!

**Comment by musjj** [comment]
I have this problem too. I tried `mode: "onChange"` and `mode: "all"`, but it doesn't work. Did you managed to solve this?

**Comment by bflemi3** [comment]
Is there no resolution to this?

**Comment by chardin1** [comment]
This would be really useful. What's the point of `watch` in a validation library if you can't tell if the value is valid?
